### PR TITLE
Fix: Catch an exception when a http failure occurs

### PR DESCRIPTION
**Pull request by RikudouSage**
# Description

Fixes an error reported on Slack where a http exception isn't caught.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] Unit tests

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Metrics/DefaultMetricsSender.php [hunk 1]
```diff
@@ -33,6 +34,11 @@ public function sendMetrics(MetricsBucket $bucket): void
         foreach ($this->configuration->getHeaders() as $name => $value) {
             $request = $request->withHeader($name, $value);
         }
-        $this->httpClient->sendRequest($request);
+
+        try {
+            $this->httpClient->sendRequest($request);
+        } catch (ClientExceptionInterface) {
+            // ignore the error
```
Comment (sighphyre) [inline src/Metrics/DefaultMetricsSender.php:41]: This looks intentionally ignored but do we not want to log this? Totally okay with the answer being no here

**Review by sighphyre — APPROVED** [review]
LGTM!

File: src/Metrics/DefaultMetricsSender.php (hunk 1 above)
Comment (RikudouSage) [reply src/Metrics/DefaultMetricsSender.php:41]: It doesn't currently log anything as there's no logging set up, not sure how to log this, the only thing I can come up with is [`error_log()`](https://www.php.net/manual/en/function.error-log.php). The SDK would probably need some logger support in general, not sure it's worth creating a partial support here.

Comment (sighphyre) [reply src/Metrics/DefaultMetricsSender.php:41]: No sounds super sensible to me!

**Pull request merged**